src/transforms/levels.jl

```diff
@@ -0,0 +1,67 @@
+"""
+    Levels(:a => ["yes,"no"])
+
+Return a copy of the table with specified levels and orders for categorical columns
+allowing only changing the order of the column.
+
+# Examples
+
+```julia
+Levels(:a => ["yes, "no"], :c => [1, 2, 4], :d => ["a", "b", "c"])
+Levels("a" => ["yes", "no"], "c" => [1, 2, 4], ordered = ["a", "c"])
+Levels(:a => ["yes", "no"], :c => [1, 2, 4], :d => [1, 23, 5, 7], ordered = [:a, :b, :c])
+```
+"""
+struct Levels{K} <: Stateless
+  levelspec::K
+  ordered::Vector{Symbol}
+end
+
+Levels(pairs::Pair{Symbol}...; ordered=Symbol[]) =
+  Levels(NamedTuple(pairs), ordered)
+
+Levels(pairs::Pair{K}...; ordered=K[]) where {K <: AbstractString} =
+  Levels(NamedTuple(Symbol(k) => v for (k,v) in pairs), Symbol.(ordered))
+
+isrevertible(transform::Levels) = true
+
+# when the col is already a categorical array and wanna change levels and order
+categorify(l::AbstractVector, x::CategoricalVector, o) = categorical(x, levels=l, ordered=o), levels(x)
+
+# when the col is normal array and want to change to categorical array
+categorify(l, x::AbstractVector, o) = categorical(x, levels=l, ordered=o), unwrap 
+# when the col is not need for change or convert back to normal array
+categorify(f::Function, x::AbstractVector, o) = o ? (categorical(x, ordered=true), levels(x)) : (f.(x), f)
+
+function apply(transform::Levels,table)
+  cols = Tables.columns(table)
+  names = Tables.columnnames(cols)
+  newlevels = transform.levelspec
```

Comment:
Notice how you are assigning a variable called levelspec to a variable newlevels. Please pick a single name. Given that this variable is only used one time, I would not even bother creating a new variable inside the function. You can access it directly in the get(transform.levelspec, nm, identity) call.